Add the python files check for prepush script

RonEld opened the pull request:
## Description
Add the `check-python-files.sh` test to the prepush git script.
This should avoid errors in the python scripts, prior to making the PR.


## Status
**READY**

## Requires Backporting
NO  


## Additional comments
Requires `pylint` to be installed, using `pip install pylint`

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported


## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

andresag01 commented on tests/git-scripts/README.md:
```diff
@@ -6,6 +6,11 @@ for more information, see the [git documentation](https://git-scm.com/docs/githo
+Prerequisites for running the scripts:
+* python
+* perl
+* pylint (installed with `pip install pylint`)
```
Does it require a specific python or pylint version? I remember there was a disagreement/issue with this at some point

RonEld replied:
It does require some pylint version, not sure which though...
When I installed pylint through  aptitude (`apt-get install pylint`) it was an old version. This is why it should be installed through `pip`, which installs a later version

RonEld commented:
I have rebased to resolve conflicts.
I have also added `doxygen` to the prerequisite list, as the doxygen file checker script was added recently as well..

RonEld commented:
Note for gatekeepr and reviewer: 
Before merging, need to consider https://github.com/ARMmbed/mbedtls/issues/2348

RonEld commented:
Closing this PR in favor of #2639
